src/main/java/com/atlassian/bitbucket/jenkins/internal/provider/JenkinsAuthWrapper.java

```diff
@@ -0,0 +1,14 @@
+package com.atlassian.bitbucket.jenkins.internal.provider;
+
+import com.google.inject.ImplementedBy;
+import org.acegisecurity.Authentication;
+
+/**
```

Comment:
We already have a `DefaultJenkinsProvider` for this purpose. You could just add the `getAuthentication` method to that so we can use a single mocking class rather than two.

Reply:
I have a different take on this. Jenkins class is not a very readable class and break lots of design principles. We have 2 options,

1. We either expand all Jenkins class method in JenkinsProvider instead of one method returning Jenkins instance (I prefer this one)
2. We introduce responsibility specific class like one did here. This has the advantage of making dependency explicitly clear. For example `AuthorizeConfirmationConfig` depends on Jenkins authentication. 

Having a method in the same class that returns authentication and Jenkins instance makes the dependent class transitive dependency slightly confusing. Any further modification in that class could use Jenkins instance instead of a specific authentication method without realizing. We either go for solution 1 or solution 2. I have tried to use the second one.

Reply:
Ah that's interesting, I hadn't thought about it from that perspective. I agree, mocking individual methods seems like an elegant way to better understand class dependencies (although for some Jenkins methods we call, like `getDescriptorOrDie` or `getAllItems`, this doesn't clarify much)
A quick browse over the repo suggests we call 6 or 7 different methods, so these would also need their own module. A decision on this feels OOS for this PR but removing Jenkins dependencies from our unit tests is something I'm keen to do so we should decide on an approach. Maybe share it in the room?